### ISSUE TITLE
Make sneaking toggleable

### ISSUE THREAD

**Issue by Kalabasa** [luanti-org/luanti#2194]
Right now, you have to hold down the sneak button to sneak. This can be painful (especially for touch UI). I think sneaking should be toggled (activated or deactivated) by the keypress instead.

**Comment by nanepiwo** [comment]
I disagree. Sneaking should be like fast, having to hold down a key. On Android builds, there should be a key that toggles sneak, but not on computers.

**Comment by Kalabasa** [comment]
How about making it an option™ in the controls screen?
I fell too many times while building because of not holding the button hard enough (which is painful).

**Comment by nanepiwo** [comment]
That would be plausible, as an option in the keymapping settings that toggles sneak and defaults to nothing.

**Comment by paramat** [comment]
We have new sneak code now, many players are used to old sneak code as it is and do not want any change.
@SmallJoker @sfan5 any opinion on this for new sneak code?

**Comment by SmallJoker** [comment]
I quite like the idea of making this a touch UI-only feature. Holding sneak isn't a problem if there's a physical keyboard.

**Comment by paramat** [comment]
I quite like the idea of toggle, i find holding sneak uncomfortable even with a custom keymap. I've tried the normal keymap of using shift and it is painful to use your little finger for long perioids.
It would need to be a separate key though as many prefer a hold-down key.

**Comment by paramat** [comment]
I don't care about touch interfaces, they're inherently bad anyway.
Is this worth keeping open?
Is anyone interested in coding this sometime? if not closing in 1 month.

**Comment by paramat** [comment]
I disagree with toggling instead, but an additional control or an option may be ok.
Holding sneak is still an issue with a keyboard becasue of finger stress.

**Comment by Ezhh** [comment]
You don't need to hold it with new movement code (I much prefer when you **do** need to hold it). 

This being an option for new movement would make me hate it less, since the change in controls is one of my biggest issues with.

**Comment by paramat** [comment]
This isn't about holding sneak at an edge and releasing to drop off the edge, which i know you like, it's about general usage of the key.

**Comment by Ezhh** [comment]
But there is really no other need to hold it...

**Comment by paramat** [comment]
I think so, when you're moving near and edge you need to hold it :)

**Comment by Ezhh** [comment]
So it's about holding sneak at the edge... which you shouldn't need to do much at all.

**Comment by paramat** [comment]
Mostly it may be up to 10s but in some situations a player may hold it for a minute or more. Being the pinky finger that makes it more difficult to hold. Not a big issue though i agree.

**Comment by Xeno333** [comment]
Do we have an intention on doing this? If so I might give it a try.

**Comment by appgurueu** [comment]
Yes, this issue has the "concept approved" label. Feel free to give it a try.
